### PR TITLE
Add workflow building the app with its deps fetched through seedwing-proxy

### LINKED ISSUE #1: Create an example CI job using seedwing-proxy
Would be nice to add another job to the java-ci GitHub Actions workflow that:

* Starts seedwing-policy-server
* Starts seedwing-proxy
* Configures maven config
* Builds the same example java project as the existing workflow

### PR DESCRIPTION
**Pull request by jcrossley3**
Fixes #1

### CODE REVIEW

**Review by lulf — APPROVED** [review]
Nice! Would be great if you could add a reference to this on the main repo README as well.

I originally thought to put the job in the java-ci.yaml workflow (and having corresponding flows for other languages eventually), but now I'm not sure anymore. Let's just keep it like this until we add the proxy to the github action.

File: policies/vulnerabilities.dog [hunk 1]
```diff
@@ -1,2 +1,10 @@
-/// For a given CycloneDX SBOM, check if any of the component package URLs for vulnerabilities and produce an OpenVEX document for it.
-pattern not-affected = cyclonedx::component-purls | lang::map<uri::purl> | osv::from-purl | openvex::from-osv | list::none<openvex::affected>
+/// For a given collection of purls, check for vulnerabilities and produce an OpenVEX document for it.
+pattern not-affected = lang::map<uri::purl> | osv::from-purl | openvex::from-osv | openvex::not-affected
```
Comment (lulf) [inline policies/vulnerabilities.dog]: For consistency, maybe purl-not-affected ?

**Review by lulf — APPROVED** [review]

**Comment by lulf** [comment]
@jcrossley3 If this is ready to merge, could you resolve the conflicts and re-push?

**Comment by jcrossley3** [comment]
> @jcrossley3 If this is ready to merge, could you resolve the conflicts and re-push?

Done. I'm still not sure I love it, though. I feel like I need to affirm the output from the proxy before I update the README.

**Review by lulf — APPROVED** [review]

**Pull request merged**